File: SakhaTyla.Web.Ng/src/app/article-core/article.model.ts
import { OrderDirection } from '../core/models/order-direction.model';
import { PageSettings } from '../core/page.model';
import { ArticleTag } from '../article-tag-core/article-tag.model';
import { ArticleFilter } from './article-filter.model';

export class Article {
  id: number;
  creationDate: Date;
  modificationDate: Date;
  creationUserId: number;
  modificationUserId: number;
  title: string;
  text: string;
  textSource: string;
  fromLanguageId: number;
  toLanguageId: number;
  fuzzy: boolean;
  categoryId: number;
  tags: ArticleTag[];
}

export class ArticleListState {
  pageSize = PageSettings.pageSize;
  pageIndex = 0;
  filter = new ArticleFilter();
  orderBy?: string;
  orderDirection?: OrderDirection;
}

File: SakhaTyla.Core/Requests/Articles/Models/ArticleModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;

namespace SakhaTyla.Core.Requests.Articles.Models
{
    public class ArticleModel
    {
        public ArticleModel(string title, string text, string textSource)
        {
            Title = title;
            Text = text;
            TextSource = textSource;
        }

        [DisplayName("Id")]
        public int Id { get; set; }

        [DisplayName("Creation Date")]
        public DateTime CreationDate { get; set; }

        [DisplayName("Modification Date")]
        public DateTime ModificationDate { get; set; }

        [DisplayName("Creation User")]
        public int? CreationUserId { get; set; }
        public Users.Models.UserShortModel? CreationUser { get; set; }

        [DisplayName("Modification User")]
        public int? ModificationUserId { get; set; }
        public Users.Models.UserShortModel? ModificationUser { get; set; }

        [DisplayName("Title")]
        public string Title { get; set; }

        [DisplayName("Text")]
        public string Text { get; set; }

        [DisplayName("Text Source")]
        public string TextSource { get; set; }

        [DisplayName("From Language")]
        public int FromLanguageId { get; set; }
        public Languages.Models.LanguageShortModel FromLanguage { get; set; } = null!;

        [DisplayName("To Language")]
        public int ToLanguageId { get; set; }
        public Languages.Models.LanguageShortModel ToLanguage { get; set; } = null!;

        [DisplayName("Fuzzy")]
        public bool Fuzzy { get; set; }

        [DisplayName("Category")]
        public int? CategoryId { get; set; }
        public Categories.Models.CategoryShortModel? Category { get; set; }
    }
}
